### PR TITLE
Add roadmap page

### PR DESCRIPTION
**Pull request by mooreryan**
See #81 and #82.

Just brainstorming some ideas about how the roadmap page could look.

### CODE REVIEW

**Comment by mooreryan** [comment]
I set it as a draft for now...more as a starting point for discussion.

File: docs/roadmap.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Roadmap
+
+Welcome to the roadmap for Melange! While Melange is now at version 1.0, our team is actively developing and improving it to create the best OCaml experience on JavaScript.
```
Comment (jchavarri) [inline docs/roadmap.md:3]: Maybe this section can start just with the sentence below ("This page provides...")

File: docs/roadmap.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+This page provides an overview of our development goals and tasks for each quarter. Our roadmap is designed to give you a clear understanding of what's coming next and the features and improvements we're working on.
```
Comment (jchavarri) [inline docs/roadmap.md:5]:
```suggestion
This page provides an overview of the development goals and tasks for each quarter. The roadmap is designed to give a clear understanding of what's coming next and the features and improvements under development.
```

**Review by jchavarri — COMMENTED** [review]
Thanks! I added some suggestions, curious what @anmonteiro thinks.

**Comment by mooreryan** [comment]
Thanks for the comments.  I pushed commits resolving the simpler ones.

File: docs/roadmap.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+Please note that the roadmap is a rough outline and is subject to change. We encourage you to provide feedback, suggestions, and feature requests through our communication channels.
+
+## Q2 2023
+
+- [Roadmap](https://docs.google.com/document/d/1279euT9LeJIkwAUYqazqeh2lc8c7TLQap2_2vBNcK4w/)
```
[the remark this answers is not in the document]
Comment (anmonteiro) [reply docs/roadmap.md:9]: I think it's probably fine to include the documents for now. I suspect the content might get repeated and too long for people to read?

File: docs/rationale.md [hunk 4]
```diff
@@ -119,10 +119,9 @@ Melange.
...
-tool for OCaml developers targeting the JavaScript platform.
-The [Q2 2023
-roadmap](https://docs.google.com/document/d/1279euT9LeJIkwAUYqazqeh2lc8c7TLQap2_2vBNcK4w/)
-includes short and long term goals for the project.
+While reaching v1.0 marks a major milestone for Melange, it is only
+the beginning of the journey. The Melange team remains committed to
+continuously improving Melange, ensuring it remains a robust and
+efficient tool for OCaml developers targeting the JavaScript platform.
+The [roadmap](roadmap.md) includes short and long term goals for the
+project.
```
Comment (anmonteiro) [inline docs/rationale.md:127]:
```suggestion
The [roadmap](roadmap.md) page details past work and
current goals of Melange.
```

File: docs/roadmap.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# Roadmap
+
+This page provides an overview of the development goals and tasks for each quarter. The roadmap is designed to give a clear understanding of what's coming next and the features and improvements under development.
```
Comment (anmonteiro) [inline docs/roadmap.md:3]:
```suggestion
This page contains:

- a historical overview of Melange development
- details of the Melange roadmap, offering a clear view into the current priorities and features being worked on.
```

**Comment by mooreryan** [comment]
I will mark this as "not-draft".  Just let me know if you want me to clean up and inline the roadmap.

**Review by jchavarri — APPROVED** [review]
Thanks for adding this page :) this looks good to me, let's see what @anmonteiro thinks before merging.

**Comment by anmonteiro** [comment]
Thanks

**Pull request merged**